include/fi_signal.h

```diff
@@ -57,6 +58,33 @@ struct fd_signal {
 	int		fd[2];
 };
 
+static inline int fi_close_fd(int fd)
+{
+#ifndef _WIN32
+	return close(fd);
+#else
+	return closesocket(fd);
+#endif
+}
+
+static inline ssize_t fi_write_fd(int fd, const void *buf, size_t count)
+{
+#ifndef _WIN32
+	return write(fd, buf, count);
+#else
+	return send(fd, buf, count, 0);
+#endif
+}
+
+static inline ssize_t fi_read_fd(int fd, void *buf, size_t count)
+{
+#ifndef _WIN32
+	return read(fd, buf, count);
+#else
+	return recv(fd, buf, count, 0);
+#endif
```

Comment:
ifdefs make code unreadable,  Please keep definitions to osd.h files.  I think the best option here would be to replace read/write with send/recv, since those calls work on all platforms.  I don't think there's a good solution for close socket..  maybe use closesocket in the code and define that for all non-windows platforms?  close_fd reads like it closes any fd, but it only works with sockets.